src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3273,13 +3273,21 @@ export class ResultsViewPageStore {
     readonly mutationsByGene = remoteData<{
         [hugoGeneSymbol: string]: Mutation[];
     }>({
-        await: () => [
-            this.selectedMolecularProfiles,
-            this.defaultOQLQuery,
-            this.mutationsReportByGene,
-            this.filteredSampleKeyToSample,
-            this.structuralVariantsReportByGene,
-        ],
+        await: () =>
+            this.hideUnprofiledSamples
```

Comment:
@alisman @adamabeshouse guessing this is not the canonical way to do this, lmk what you recommend

Reply:
basically the `filteredSampleKeyToSample` is only necessary when the `hideUnprofiledSamples` item is selected in the UI: https://github.com/cBioPortal/cbioportal-frontend/pull/3266

Reply:
this is basically what I have done in the past. the only change I'd make to this is to make it clear what is contingent. so 
```
const promises = [this.selectedMolecularProfiles, ... , this.structuralVariantsReportByGene];
if (this.hideUnprofiledSamples) {
promises.push(this.filteredSampleKeyToSample);
}
return promises;
```

Reply:
i see - yes that's cleaner